server/asset/eth/coiner.go

```diff
@@ -176,109 +173,104 @@ func (backend *Backend) newSwapCoin(coinID []byte, contractData []byte, sct swap
-
-	case sctInit:
-		// Uninitiated state is zero confs. It could still be in mempool.
-		// It is important to only trust confirmations according to the
-		// swap contract. Until there are confirmations we cannot be sure
-		// that initiation happened successfully.
-		if dexeth.SwapStep(swap.State) == dexeth.SSNone {
-			// Assume the tx still has a chance of being mined.
-			return 0, nil
-		}
-		// Any other swap state is ok. We are sure that initialization
-		// happened.
-
-		// The swap initiation transaction has some number of
-		// confirmations, and we are sure the secret hash belongs to
-		// this swap. Assert that the value, receiver, and locktime are
-		// as expected.
-		value, err := dexeth.ToGwei(new(big.Int).Set(swap.Value))
-		if err != nil {
-			return -1, fmt.Errorf("unable to convert value: %v", err)
-		}
-		if value != c.value {
-			return -1, fmt.Errorf("expected swap val (%dgwei) does not match expected (%dgwei)",
-				c.value, value)
-		}
-		if swap.Participant != c.counterParty {
-			return -1, fmt.Errorf("expected swap participant %q does not match expected %q",
-				c.counterParty, swap.Participant)
-		}
-		if !swap.RefundBlockTimestamp.IsInt64() {
-			return -1, errors.New("swap locktime is larger than expected")
-		}
-		locktime := swap.RefundBlockTimestamp.Int64()
-		if locktime != c.locktime {
-			return -1, fmt.Errorf("expected swap locktime (%d) does not match expected (%d)",
-				c.locktime, locktime)
-		}
+	// Uninitiated state is zero confs. It could still be in mempool.
+	// It is important to only trust confirmations according to the
+	// swap contract. Until there are confirmations we cannot be sure
+	// that initiation happened successfully.
+	if swap.State == dexeth.SSNone {
+		// Assume the tx still has a chance of being mined.
+		return 0, nil
+	}
+	// Any other swap state is ok. We are sure that initialization
+	// happened.
+
+	// The swap initiation transaction has some number of
+	// confirmations, and we are sure the secret hash belongs to
+	// this swap. Assert that the value, receiver, and locktime are
+	// as expected.
+	if swap.Value != c.init.Value {
+		return -1, fmt.Errorf("tx data swap val (%dgwei) does not match contract value (%dgwei)",
+			c.init.Value, swap.Value)
+	}
+	if swap.Participant != c.init.Participant {
+		return -1, fmt.Errorf("tx data participant %q does not match contract value %q",
+			c.init.Participant, swap.Participant)
+	}
+
+	// locktime := swap.RefundBlockTimestamp.Int64()
+	if !swap.LockTime.Equal(c.init.LockTime) {
+		return -1, fmt.Errorf("expected swap locktime (%s) does not match expected (%s)",
+			c.init.LockTime, swap.LockTime)
+	}
+
+	bn, err := c.backend.node.blockNumber(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("unable to fetch block number: %v", err)
+	}
+	return int64(bn - swap.BlockHeight + 1), nil
+}
 
-		bn, err := c.backend.node.blockNumber(c.backend.rpcCtx)
+func (c *redeemCoin) Confirmations(ctx context.Context) (int64, error) {
```

Comment:
Do we ever check redeem confs?  I'm drawing a blank on this for some reason

Reply:
I don't think so. I put added a `TODO` in the `(*Backend).newRedeemCoin` docs about it.